reproducibility_project/src/engines/mcccs/project.py

```diff
@@ -72,33 +100,76 @@ def has_fort_files(job):
 
 @Project.label
 @Project.pre(
-    lambda j: j.sp.simulation_engine == "mcccs" and j.sp.molecule == "methane"
+    lambda j: j.sp.engine == "mcccs" and j.sp.molecule == ("methaneUA")
 )
 def files_ready(job):
     """Check if the keywords in the fort.4 files have been replaced."""
     # Link: https://stackoverflow.com/questions/32749350/check-if-a-string-is-in-a-file-with-python
-    job.doc.files_ready = False
-    file_names = ["melt", "cool", "equil", "prod"]
-    print(job.ws)
-    keywords = ["NCHAIN", "LENGTH", "TEMPERATURE", "PRESSURE", "VARIABLES"]
-    c = 0
-    for name in file_names:
-        file_name = job.ws + "/fort.4." + name
-        i = 0
-        for i in range(len(keywords)):
-            with open(file_name) as myfile:
-                if keywords[i] in myfile.read():
-                    c += 1
-    if c == 0:
-        job.doc.files_ready = True
-
-    return job.doc.files_ready
+    if job.doc.ensemble == "GEMC-NVT":
+        job.doc.files_ready = False
+        file_names = ["melt", "cool", "equil", "prod"]
+        keywords = [
+            "NCHAIN1",
+            "NCHAIN2",
+            "LENGTH1",
+            "LENGTH2",
+            "TEMPERATURE",
+            "RCUT",
+            "NCHAINTOT",
+            "INIX1",
+            "INIY1",
+            "INIZ1",
+            "INIX2",
+            "INIY2",
+            "INIZ2",
+            "VARIABLES",
+        ]
+        c = 0
+        for name in file_names:
+            file_name = job.ws + "/fort.4." + name
+            i = 0
+            if not job.isfile("fort.4." + name):
+                continue
+            for i in range(len(keywords)):
+                with open(file_name) as myfile:
+                    if keywords[i] in myfile.read():
+                        c += 1
+        if c == 0:
+            job.doc.files_ready = True
+
+        return job.doc.files_ready
+
+    if job.doc.ensemble == "NPT":
+        job.doc.files_ready = False
+        file_names = ["melt", "cool", "equil", "prod"]
+        keywords = [
+            "NCHAIN",
+            "LENGTH",
+            "TEMPERATURE",
+            "PRESSURE",
+            "RCUT",
+            "VARIABLES",
+        ]
+        c = 0
+        for name in file_names:
+            file_name = job.ws + "/fort.4." + name
+            i = 0
+            if not job.isfile("fort.4." + name):
+                continue
+            for i in range(len(keywords)):
+                with open(file_name) as myfile:
+                    if keywords[i] in myfile.read():
+                        c += 1
+        if c == 0:
+            job.doc.files_ready = True
+
+        return job.doc.files_ready
 
 
 @Project.label
 def has_restart_file(job):
     """Check if the job has a restart file."""
-    return job.isfile("fort.77")
+    return job.sp.ensemble == "GEMC-NVT" or job.isfile("fort.77")
```

Comment:
This will evaluate to true in either of these situations, is that was is expected?

Reply:
Fixed.